var/spack/repos/builtin/packages/open3d/package.py

```diff
@@ -120,11 +120,11 @@ def install(self, spec, prefix):
 
     @run_after("install")
     @on_package_attributes(run_tests=True)
-    def check_import(self):
-        if "+python" in self.spec:
-            self.run_test(
-                python.path,
-                ["-c", "import open3d"],
-                purpose="checking import of open3d",
-                work_dir="spack-test",
-            )
+    def test_open3d_import(self):
+        """Checking import of open3d"""
+        if "+python" not in self.spec:
+            raise SkipTest("Package must be installed with +python")
```

Comment:
The installation process doesn't catch `SkipTest` exceptions so this would likely result in an installation failure if the package is installed with `~python`.  This should just return (as was essentially what happened before).

```suggestion
            return
```

Reply:
The unfortunate side-effect of that is that this will be marked `PASSED` when run as a stand-alone test.